### PR TITLE
IIDM network import using alternative IIDM implementation

### PR DESCRIPTION
**Pull request by geofjamg**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
#813 

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Feature

**What is the current behavior?** *(You can also link to an open issue here)*
See issue

**What is the new behavior (if this is a feature change)?**
See issue

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [x] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*
There is no API break. Some methods have been deprecated.

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/Conversion.java [hunk 1]
```diff
@@ -158,7 +164,7 @@ private Network createNetwork() {
         profiling.start();
         String networkId = cgmes.modelId();
         String sourceFormat = "CGMES";
-        Network network = NetworkFactory.create(networkId, sourceFormat);
+        Network network = networkFactory.createNetwork(networkId, sourceFormat);
```
Comment (mathbagu) [inline cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/Conversion.java]: I think you should keep create as method name: a NetworkFactory is a class that creates networks... there is no need to repeat what is created in the method name. What do you think?

Maybe you can’t. In that case, consider newNetwork?

File: iidm/iidm-converter-api/src/main/java/com/powsybl/iidm/import_/Importer.java [hunk 2]
```diff
@@ -55,14 +56,26 @@
      */
     boolean exists(ReadOnlyDataSource dataSource);
 
+
+    /**
+     * @deprecated Use {@link Importer##importData(ReadOnlyDataSource, NetworkFactory, Properties)} instead.
```
Comment (mathbagu) [inline iidm/iidm-converter-api/src/main/java/com/powsybl/iidm/import_/Importer.java]: Is the javadoc référence ok? (two dashes)

Comment (geofjamg) [reply iidm/iidm-converter-api/src/main/java/com/powsybl/iidm/import_/Importer.java]: Right, link was broken. Fixed.

**Comment by geofjamg** [comment]
> Could you add a small description of these changes to the migration guide? (Create a new wiki page if it doesn’t exist yet)
Done [Migration-guide-v3.0.0](https://github.com/powsybl/powsybl-core/wiki/Migration-guide-v3.0.0)

**Pull request merged**